gcc/rust/backend/rust-compile-intrinsic.cc

```diff
@@ -1174,5 +1182,50 @@ expect_handler_inner (Context *ctx, TyTy::FnType *fntype, bool likely)
   return fndecl;
 }
 
+static tree
+assume_handler (Context *ctx, TyTy::FnType *fntype)
+{
+  // TODO: make sure this is actually helping the compiler optimize
+
+  rust_assert (fntype->get_params ().size () == 1);
+  rust_assert (fntype->param_at (0).second->get_kind ()
+	       == TyTy::TypeKind::BOOL);
+
+  tree lookup = NULL_TREE;
+  if (check_for_cached_intrinsic (ctx, fntype, &lookup))
+    return lookup;
+
+  auto fndecl = compile_intrinsic_function (ctx, fntype);
+
+  // TODO: make sure these are necessary
+  TREE_READONLY (fndecl) = 0;
+  DECL_DISREGARD_INLINE_LIMITS (fndecl) = 1;
+  DECL_ATTRIBUTES (fndecl) = tree_cons (get_identifier ("always_inline"),
```

Comment:
we should be doing this for all our intrinsics can  you add this to thee compile_intrinsic_function in another PR. We probably should avoid setting disregard inline limits for all of them though.

Reply:
the always inline attrib is what i mean should be added for all of them